library/std/src/pipe.rs

```diff
@@ -1,20 +1,66 @@
-//! Module for anonymous pipe
+//!  A cross-platform anonymous pipe.
 //!
-//! ```
-//! #![feature(anonymous_pipe)]
+//! This module provides support for anonymous OS pipes, like [pipe] on Linux or [CreatePipe] on
+//! Windows, which can be used as synchronous communication channels between related processes.
+//!
+//! # Behavior
+//!
+//! A pipe can be thought of as a bounded, interprocess [`mpsc`](crate::sync::mpsc), provided by
+//! the OS, with a platform-dependent capacity. In particular:
+//!
+//! * A read on a [`PipeReader`] blocks until the pipe is non-empty.
+//! * A write on a [`PipeWriter`] blocks when the pipe is full.
+//! * When all copies of a [`PipeWriter`] are closed, a read on the corresponding [`PipeReader`]
+//!   returns EOF.
+//! * [`PipeReader`] can be shared through copying the underlying file descriptor, but only one
+//!   process will consume the data in the pipe at any given time.
+//!
+//! # Capacity
+//!
+//! Pipe capacity is platform-dependent. To quote the Linux [man page]:
+//!
+//! > Different implementations have different limits for the pipe capacity. Applications should
+//! > not rely on a particular capacity: an application should be designed so that a reading process
+//! > consumes data as soon as it is available, so that a writing process does not remain blocked.
 //!
+//! # Examples
+//!
+//! ```no_run
+//! #![feature(anonymous_pipe)]
 //! # #[cfg(miri)] fn main() {}
 //! # #[cfg(not(miri))]
+//! # use std::process::Command;
+//! # use std::io::{Read, Write};
 //! # fn main() -> std::io::Result<()> {
```

Comment:
I don't think this is accurate, it looks like `#[cfg(not(miri))]` is getting applied only to `use std::process::Command` but not to the function itself, so Miri will see two `main`s.

You can just move the imports within `fn main()`. This needs to be done for the other examples too.